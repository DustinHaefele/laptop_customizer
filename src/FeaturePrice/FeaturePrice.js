import React from "react";


function FeaturePrice(props) {
  
    return (
      <div className="summary__option" key={props.myKey}>
        <div className="summary__option__label">{props.myKey} </div>
        <div className="summary__option__value">
          {props.name}
        </div>
        <div className="summary__option__cost">
          {new Intl.NumberFormat("en-US", {
            style: "currency",
            currency: "USD"
          }).format(props.cost)}
        </div>
      </div>
    );
}

export default FeaturePrice;