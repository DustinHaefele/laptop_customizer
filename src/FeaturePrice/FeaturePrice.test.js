import React from "react";
import ReactDOM from "react-dom";
import Enzyme from "enzyme";
import toJson from "enzyme-to-json";
import FeaturePrice from "./FeaturePrice";

describe("Feature Price Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <FeaturePrice
        myKey={'my Key'}
        cost={700}
        name={'My Prop Name'}
      />, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected',()=>{
    const wrapper = Enzyme.shallow(<FeaturePrice
      myKey={'my Key'}
      cost={700}
      name={'My Prop Name'}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
