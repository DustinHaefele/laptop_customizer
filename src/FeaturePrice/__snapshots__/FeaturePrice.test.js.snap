// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature Price Component renders UI as expected 1`] = `
<div
  className="summary__option"
  key="my Key"
>
  <div
    className="summary__option__label"
  >
    my Key
     
  </div>
  <div
    className="summary__option__value"
  >
    My Prop Name
  </div>
  <div
    className="summary__option__cost"
  >
    $700.00
  </div>
</div>
`;
